signac/synced_collections/backends/collection_json.py

```diff
@@ -0,0 +1,395 @@
+# Copyright (c) 2020 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+"""Implements a JSON SyncedCollection backend."""
+
+import errno
+import json
+import os
+import uuid
+import warnings
+from typing import Tuple
+
+from .. import SyncedAttrDict, SyncedCollection, SyncedList
+from ..buffers.memory_buffered_collection import SharedMemoryFileBufferedCollection
+from ..buffers.serialized_file_buffered_collection import (
+    SerializedFileBufferedCollection,
+)
+from ..utils import SyncedCollectionJSONEncoder
+from ..validators import json_format_validator
+
+
+# TODO: This method should be removed in signac 2.0.
```

Comment:
Yes, I'm also in favor of that. It would also allow us to do things like remove attribute-based dict access in synced_collections and only support it in internal signac classes.